extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js

```diff
@@ -147,12 +185,18 @@ class ReplicationStatusProcessor {
             task = new UpdateReplicationStatus(this);
         }
         if (task) {
-            return this.taskScheduler.push({ task, entry: sourceEntry },
-                                           sourceEntry.getCanonicalKey(),
-                                           done);
+            return this._setFailedKeys(sourceEntry, kafkaEntry, err => {
+                if (err) {
+                    this.logger.error('error setting redis hash key', {
+                        error: err,
+                    });
+                }
+                return this.taskScheduler.push({ task, entry: sourceEntry },
+                    sourceEntry.getCanonicalKey(), done);
+            });
         }
-        this.logger.warning('skipping unknown source entry',
-                            { entry: sourceEntry.getLogInfo() });
+        this.logger.warn('skipping unknown source entry',
```

Comment:
This change is unrelated to the PR, but figured I can update it along the way. 😄